File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(Progresses, Speeds):

    x = []
    for p,s in zip(Progresses, Speeds):
        
        if (100-p)%s == 0:
            x.append((100-p)//s)
        else:
            x.append((100-p)//s+1)
            
    temp = []      
    temp.append(x[0])
    
    answer = []
    
    for i in range(1, len(x)):
        
        if max(temp) < x[i]:
            answer.append(len(temp))
            while temp:
                temp.pop()
            temp.append(x[i])
        else:
            temp.append(x[i])
    
    answer.append(len(temp))
    
    return answer
    
    
    

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        Deque<Integer> lastDay = new ArrayDeque<>();
        
        for (int i=0; i<progresses.length; i++) {
            if ((100-progresses[i])%speeds[i]>0) {
                lastDay.addLast((100-progresses[i])/speeds[i] + 1);
            } else {
                lastDay.addLast((100-progresses[i])/speeds[i]);
            }
        }
        
        List<Integer> answers = new ArrayList<>();
            
        int maxDay = lastDay.pollFirst();
        int temp = 1;
        
        while (!lastDay.isEmpty()) {
            int nextDay = lastDay.pollFirst();
            
            if (maxDay < nextDay) {
                answers.add(temp);
                temp=1;
                maxDay = nextDay;
            } else {
                temp +=1;
            }
        }
        
        answers.add(temp);
        
        return answers.stream()
            .mapToInt(Integer::intValue)
            .toArray();
    }
}